substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/SubstrateJVM.java

```diff
@@ -723,14 +716,14 @@ protected void operate() {
 
             /* No further JFR events are emitted, so free all JFR-related buffers. */
```

Comment:
I think this comment might not be true anymore. We flush all the local buffers to the global buffers, but the java local buffers do not get freed here only retired.